src/job/Job.ts

```diff
@@ -1,47 +1,105 @@
-import { WithoutId } from 'mongodb';
+import humanInterval from 'human-interval';
+import { Result, err, ok } from 'neverthrow';
+import { parseExpression } from 'cron-parser';
 
-import { CronSchedule, Handler, IntervalSchedule, MomoJob, isIntervalSchedule } from './MomoJob';
-import { JobEntity } from '../repository/JobEntity';
+import { CronSchedule, Handler, IntervalSchedule, MomoJob, TypedMomoJob, isCronJob } from './MomoJob';
+import { momoError } from '../logging/error/MomoError';
 
-export type MomoJobStatus = WithoutId<JobEntity>;
+export interface ParsedIntervalSchedule extends Required<IntervalSchedule> {
+  parsedInterval: number;
+  parsedFirstRunAfter: number;
+}
 
-export interface JobDefinition {
+export interface JobDefinition<Schedule = ParsedIntervalSchedule | CronSchedule> {
```

Comment:
Rename generic to `JobSchedule`? We already use Schedule quite prominently in the public API.